src/Dialect/Krnl/Krnl.td

```diff
@@ -180,26 +180,52 @@ def KrnlSeqExtractOp : Op<Krnl_Dialect, "seqextract", [MemRefsNormalizable,
     DeclareOpInterfaceMethods<AllocationOpInterface, ["buildDealloc", "buildClone"]>]> {
   let summary = "Krnl load from a seq";
   let description = [{
-    sequence is represented with memref<memref<>>.
-    This op loads a tensor for the sequence 'seq' at position 'index',
-    and return the tensor, which will be freed by Bufferization::Deallocation.
-    The element in the sequence will become null.
+    This op loads an element from the input sequence 'seq' at position 'index'.
+    The position value is guaranteed to be positive.
+    When the attribute 'copy' is 0: the extracted element is directly returned
+    When the attribute 'copy' is 1: the extracted element is copied and then return. 
+    The returned element is regarded as allocated by this Op because the 
+    element is created (copied) when inserted into the sequence.
+    Attribute 'copy' can be 0 when analysis or code generation guarantees 
```

Comment:
I would expand a bit more on this. "If an optimization pass can determine that each element is only extracted once, then the copy=false can be used"

Is it required that each element be extracted once? If one is never extracted, should it still work? If so, "each element is extracted at most once"

Reply:
`every element is extracted exactly once`

Reply:
Technically, if an element is not used out of a list, why would it disable the `copy=0`. IMO, the correct statement is "at most once"

Reply:
The element needs to be extracted or erased exactly once from the sequence. Otherwise, the memory for the element will be leaked.

Reply:
If an element is not used, you imply that copy cannot be zero which makes no sense either.

Reply:
Additionally, if a value is used exactly twice, would it make sense to have the first use with copy=1 but the second/last use with copy=0?

Reply:
yes

Reply:
> If an element is not used, you imply that copy cannot be zero which makes no sense either.

The seqextract is a use the element.

Reply:
> If an element is not used, you imply that copy cannot be zero which makes no sense either.

An element can be accessed only through KrnlSeqExtract. If there is no KrnlSeqExtract, there is no copy attribute.